From 813074591041af6d7a6c706de60c533aa41188e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=A9bastien=20Blin?=
 <sebastien.blin@savoirfairelinux.com>
Date: Tue, 30 Mar 2021 15:19:46 -0400
Subject: [PATCH] previewrenderer: disconnect signals on destroy

also fix some build warnings

Change-Id: Ibbf2e28cb64800e65b8712c6d733aea9aa69c916
GitLab: #356
---
 src/contactadapter.h    |  1 +
 src/previewrenderer.cpp | 12 ++++++++----
 src/previewrenderer.h   |  2 ++
 3 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/src/contactadapter.h b/src/contactadapter.h
index f0e7729..7aaf312 100644
--- a/src/contactadapter.h
+++ b/src/contactadapter.h
@@ -61,6 +61,7 @@ public:
         if (filterPredicate_) {
             return filterPredicate_(index, filterRegExp());
         }
+        return false;
     }
 
 private:
diff --git a/src/previewrenderer.cpp b/src/previewrenderer.cpp
index e6ad1f8..893fbd8 100644
--- a/src/previewrenderer.cpp
+++ b/src/previewrenderer.cpp
@@ -117,13 +117,16 @@ PhotoboothPreviewRender::PhotoboothPreviewRender(QQuickItem* parent)
 {
     connect(this, &PreviewRenderer::lrcInstanceChanged, [this] {
         if (lrcInstance_)
-            connect(lrcInstance_->renderer(), &RenderManager::previewRenderingStopped, [this]() {
-                emit hideBooth();
-            });
+            rendererStoppedConnection_ = connect(lrcInstance_->renderer(),
+                                                 &RenderManager::previewRenderingStopped,
+                                                 [this]() { emit hideBooth(); });
     });
 }
 
-PhotoboothPreviewRender::~PhotoboothPreviewRender() {}
+PhotoboothPreviewRender::~PhotoboothPreviewRender()
+{
+    disconnect(rendererStoppedConnection_);
+}
 
 QString
 PhotoboothPreviewRender::takePhoto(int size)
@@ -133,6 +136,7 @@ PhotoboothPreviewRender::takePhoto(int size)
             previewImage->copy()
                 .scaled(size, size, Qt::KeepAspectRatioByExpanding, Qt::SmoothTransformation)));
     }
+    return {};
 }
 
 void
diff --git a/src/previewrenderer.h b/src/previewrenderer.h
index 239471e..11627d8 100644
--- a/src/previewrenderer.h
+++ b/src/previewrenderer.h
@@ -82,4 +82,6 @@ signals:
 
 private:
     void paint(QPainter* painter) override final;
+
+    QMetaObject::Connection rendererStoppedConnection_;
 };
-- 
2.30.2

